src/printutils.cc

```diff
@@ -14,6 +15,8 @@ std::string OpenSCAD::debug("");
 bool OpenSCAD::quiet = false;
 
 boost::circular_buffer<std::string> lastmessages(5);
+std::string mainFilename("");
+std::string currentFilename("");
```

Comment:
Maintaining such state in printutils makes me a bit uneasy. I don't have a great suggestion for alternatives though.
If you don't find any other way, perhaps we could add a comment here for future readers to keep this in mind?

Reply:
Taking two steps back: I could try to pass the mainfile via context/eval context.